### ISSUE TITLE
Initial product import test

### ISSUE THREAD

**Issue by BronsonQuick** [TheCodeCompany/wp-chargify#66]
As a developer I'd like to check to see what bugs we encounter with a "real" product endpoint. I've attached a zip of the plugin that I've built by doing a fresh clone of this repo and running `composer install --no-dev` so that just CMB2 comes bundled.

For a test run of products I'd like @scommins to please do the folllowing:

1. Install and activate the plugin.
2. Enter the live and production settings into Chargify Options -> Settings
3. Select Resync Products on the Chargify Options -> Product page and hit save. (Note: Currently the message will say "Nothing to update" but there is magic happening behind the scenes. I have [this](https://github.com/TheCodeCompany/wp-chargify/issues/65) issue to handle that bug).
4. Please check the "Logs" CPTs to see that requests have been fired to the product family endpoints. The first request should be for all the Product Families then then subsequent requests should be for each of the individual product.
5. Please check in the "Products" CPTs to see if the products were imported and that they contain the correct data.

[wp-chargify.zip](https://github.com/TheCodeCompany/wp-chargify/files/4783562/wp-chargify.zip)

**Comment by scommins** [comment]
Hey @BronsonQuick, I have been able to install and activate the plugin and run through the steps here.

Everything looks good, the product data is being set correctly.

I experienced one warning and have one other "issue". I'll list them here but we can discuss/set these up as separate issues if you like.

**Warning:**
When loading the options page (http://chargify.local/wp/wp-admin/admin.php?page=chargify_options) AFTER activating the plugin but BEFORE setting any API credentials, I got the following:
`Warning: Invalid argument supplied for foreach() in /vagrant/wp/wp-includes/class-wp-list-util.php on line 168`
This warning appeared for the **Chargify Products** field as it was trying list products when connecting to the API gave back an empty response.

I assume we might just want to check some condition before attempting to list the available products.

**Issue:**
The first time I entered in my API details and then loaded the options page, I noticed that the plugin automatically imported all of the products. This makes sense but it means I didn't actually have to complete **Step 3** listed on this issue. All I had to do was load http://chargify.local/wp/wp-admin/admin.php?page=chargify_options.
